### ISSUE TITLE
[Flaking Test] [sig-storage] CSI Volumes [Driver: csi-hostpath-v0] [Testpattern: Dynamic PV (default fs)] subPath should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow]

### ISSUE THREAD

**Issue by kacole2** [kubernetes/kubernetes#75196]
**Which jobs are failing**:
`ci-kubernetes-e2e-gce-master-new-downgrade-cluster` [sig-release-master-upgrade#gce-master-new-downgrade-cluster](https://testgrid.k8s.io/sig-release-master-upgrade#gce-master-new-downgrade-cluster&width=20)

**Which test(s) are failing**:
`[sig-storage] CSI Volumes [Driver: csi-hostpath-v0] [Testpattern: Dynamic PV (default fs)] subPath should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow]`

[link to prow](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-master-new-downgrade-cluster/1304)

**Since when has it been failing**:
2019-02-22 @ 20:43

**Testgrid link**:
[sig-release-master-upgrade#gce-master-new-downgrade-cluster](https://testgrid.k8s.io/sig-release-master-upgrade#gce-master-new-downgrade-cluster&width=20)

**Reason for failure**:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/subpath.go:313
Expected grep stdout to be empty (i.e. no mount found).
Expected
    <string>: overlay on /var/lib/kubelet/pods/6f95bde9-415c-11e9-b025-42010a8a0009/volumes/kubernetes.io~csi/pvc-6d0cf06d-415c-11e9-b025-42010a8a0009/mount type overlay (rw,relatime,lowerdir=/var/lib/docker/overlay2/l/TR7RSLOTG32K74YLSFHOGEF7BR:/var/lib/docker/overlay2/l/4UZHXW7X6IT7U6HHVNMMW6IPU4:/var/lib/docker/overlay2/l/3OJLNS5IZVWKQQJFPKON64H6EI,upperdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/diff,workdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/work)
    overlay on /var/lib/kubelet/pods/6f95bde9-415c-11e9-b025-42010a8a0009/volumes/kubernetes.io~csi/pvc-6d0cf06d-415c-11e9-b025-42010a8a0009/mount type overlay (rw,relatime,lowerdir=/var/lib/docker/overlay2/l/TR7RSLOTG32K74YLSFHOGEF7BR:/var/lib/docker/overlay2/l/4UZHXW7X6IT7U6HHVNMMW6IPU4:/var/lib/docker/overlay2/l/3OJLNS5IZVWKQQJFPKON64H6EI,upperdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/diff,workdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/work)
    overlay on /home/kubernetes/containerized_mounter/rootfs/var/lib/kubelet/pods/6f95bde9-415c-11e9-b025-42010a8a0009/volumes/kubernetes.io~csi/pvc-6d0cf06d-415c-11e9-b025-42010a8a0009/mount type overlay (rw,relatime,lowerdir=/var/lib/docker/overlay2/l/TR7RSLOTG32K74YLSFHOGEF7BR:/var/lib/docker/overlay2/l/4UZHXW7X6IT7U6HHVNMMW6IPU4:/var/lib/docker/overlay2/l/3OJLNS5IZVWKQQJFPKON64H6EI,upperdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/diff,workdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/work)
    overlay on /home/kubernetes/containerized_mounter/rootfs/var/lib/kubelet/pods/6f95bde9-415c-11e9-b025-42010a8a0009/volumes/kubernetes.io~csi/pvc-6d0cf06d-415c-11e9-b025-42010a8a0009/mount type overlay (rw,relatime,lowerdir=/var/lib/docker/overlay2/l/TR7RSLOTG32K74YLSFHOGEF7BR:/var/lib/docker/overlay2/l/4UZHXW7X6IT7U6HHVNMMW6IPU4:/var/lib/docker/overlay2/l/3OJLNS5IZVWKQQJFPKON64H6EI,upperdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/diff,workdir=/var/lib/docker/overlay2/23e122e3c0538f7c7f7fd17baa280334cffbd86f6198538887778f1d712cb313/work)
    
to be empty
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/utils/utils.go:255
```

**Anything else we need to know**:
/milestone v1.14
/sig storage
/kind flake
/priority important-soon

**Comment by soggiest** [comment]
Hello! We are in code freeze for 1.14. If this issue is release blocking, can we get a review on this issue? If this issue is not release blocking can we move it to 1.15?

@kubernetes/sig-storage-misc

**Comment by msau42** [comment]
We have not triaged the "gracefully deleted" test case, only the "force deleted" test case.  cc @cofyc @jingxu97 if this is the same issue or not

**Comment by msau42** [comment]
@kubernetes/sig-storage-test-failures

**Comment by mariantalla** [comment]
Hey @msau42 / @kubernetes/sig-storage-test-failures , any updates on this?

For more info: it's also had consecutive failed runs in https://testgrid.k8s.io/sig-release-1.14-all#gce-1.14-1.13-downgrade-cluster.

**Comment by msau42** [comment]
Will investigate today.

**Comment by cofyc** [comment]
The test '[sig-storage] CSI Volumes [Driver: csi-hostpath-v0] [Testpattern: Dynamic PV (default fs)] subPath should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow]' does not fail [in recent runs](https://testgrid.k8s.io/sig-release-master-upgrade#gce-master-new-downgrade-cluster&width=20&include-filter-by-regex=subPath%20should%20unmount%20if%20pod%20is%20gracefully%20deleted%20while%20kubelet%20is%20down). It probably has been fixed by PR https://github.com/kubernetes/kubernetes/pull/74652 which fixed a reconstruction issue of CSI volumes.
I will check if it starts to fail again.

**Comment by msau42** [comment]
The downgrade test is running against a 1.13 cluster.  We will need to backport the fix to 1.13 if we want to fix these flakes.

**Comment by spiffxp** [comment]
Kicking this back to under investigation, flakes appear to have reduced slightly? maybe? but it's difficult to tell
https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=grep%20stdout&job=gce&test=subPath
<img width="1117" alt="Screen Shot 2019-03-16 at 12 33 06 PM" src="https://user-images.githubusercontent.com/49258/54480624-d189d780-47e7-11e9-80b5-f193c9370925.png">

**Comment by msau42** [comment]
The fix is waiting for cherry pick to 1.13 in order to fix downgrade job flakes.

There are 3 different bugs open for this test case that are all open due to waiting for 1.13 cherrypick.  Can we consolidate them all?

**Comment by alejandrox1** [comment]
A little delayed getting on top of this but we will be looking into this again. For now, the specific test failure in this issue is no longer in place.
/close
